homeassistant/components/mqtt/entity.py

```diff
@@ -1185,6 +1185,24 @@ def device_info_from_specifications(
     return info
 
 
+@callback
+def ensure_via_device_exists(
+    hass: HomeAssistant, device_info: DeviceInfo | None, config_entry: ConfigEntry
+) -> None:
+    """Ensore the via device is in the device registry."""
+    if device_info is None:
+        return
+    device_registry = dr.async_get(hass)
+    if CONF_VIA_DEVICE in device_info and not device_registry.async_get_device(
+        identifiers={device_info["via_device"]}
+    ):
+        # Ensure the via device exists in the device registry
+        device_registry.async_get_or_create(
```

Comment:
It's somewhat odd the device registry didn't complain when `via_device` references a device which does not exist. Maybe it should at least warn?

Reply:
Yes, for sure a different PR 👍